trin-history/src/storage.rs

```diff
@@ -182,11 +182,15 @@ impl HistoryStorage {
     }
 
     fn total_entry_count(&self) -> Result<u64, ContentStoreError> {
+        let timer = self
+            .metrics
+            .start_storage_process_timer("total_entry_count");
         let conn = self.sql_connection_pool.get()?;
```

Comment:
The metric would capture the 'self.sql_connection_pool.get()' call.
My understanding is that here we are waiting to get a connection to the db. Is there a limit on number of connection and sometimes we wait here or is it usually instant?

I'm not sure this should qualify as time spent as part of this storage operation (in case db is blocked by other queries).

Reply:
There is a read write lock over storage instance. If `self.sql_connection_pool.get()`. This is important information if it is blocking then that is important information.

We want to know if these calls are slow `self.sql_connection_pool.get()` is apart of that

Reply:
Doesn't read-write lock allow multiple (unlimited?) reads at the same time? If that's the case, and `sql_connection_pool` has a limit, then we could be blocked on one request due to other requests taking long time.

And if that's desired depends on how you interpret the graphs. If there is congestion, all query metrics might go up while in fact problem can be with only one type of query.

If you time the requests without getting connection from the pool, you know how long each request takes and if there is congestion you will see exactly what is causing it. It's true that in this case you don't know how affected other queries are, so there is a trade off.

Either way, I'm fine with whatever you decide.

Reply:
The `r2d2` crate that we use [has a max number of connections](https://docs.rs/r2d2/0.8.10/r2d2/struct.Builder.html#method.max_size).

Reply:
Maybe we can report a new metric on how many connections is being used then in a future PR. If we are hitting the max number of connections and that is causing delays I think there is a problem with our code. My goal with these stats is to see how long it takes to execute the functions storage. Getting a connection to the database is included in that as that is what the code calling storage functions has to deal with.